Mailchimp tracking

jacqui opened the pull request:
Closes #436 

I setup a new MC campaign and ensured all tracking was turned on (clicks, opens, ecommerce) in the campaign builder. Only the ecommerce one wasn't already checked.

I then sent the campaign, including a link to the homepage (on vercel) and the currently featured article. @TylerFisher  you probably received this email.

I checked that both links included two query string params: mc_cid and mc_eid. This matches the MC docs, so yay.

So... this PR adds:

* setting a cookie (called it `nc_mct` for news catalyst mailchimp tracking) that stores the campaign & email ID if the query string params are present
* this currently happens on the homepage and article page
* functionality in `useAnalytics` hook
* setting custom dimension 5 to true in GA if mailchimp params are present

TylerFisher commented:
Can't we handle sending the dimension with the pageview the same way we do with reading history, in `pages/_app.js`? That way we don't have to add anything to the homepage/article page specifically, it just works across the site.

jacqui commented:
> Can't we handle sending the dimension with the pageview the same way we do with reading history, in `pages/_app.js`? That way we don't have to add anything to the homepage/article page specifically, it just works across the site.

Good point - I've moved this tracking to _app.js and removed it from the homepage and article, so it just works.
